libraries/datahandler/src/main/java/org/eclipse/sw360/datahandler/thrift/ThriftValidate.java

```diff
@@ -49,253 +36,336 @@ private ThriftValidate() {
 
         if (license.isSetObligations() && license.isSetObligationDatabaseIds()) {
-            for (Obligation oblig : license.getObligations()) {
-                license.addToObligationDatabaseIds(oblig.getId());
-            }
+            license.getObligations().forEach(oblig -> license.addToObligationDatabaseIds(oblig.getId()));
         }
         license.unsetObligations();
 
-        // Check type
         license.setType(TYPE_LICENSE);
-
-        // Unset optionals
         license.unsetPermissions();
     }
 
-
+    /**
+     * Prepares a User object for database insertion.
+     *
+     * @param user The User to prepare
+     * @throws SW360Exception if validation fails
+     */
     public static void prepareUser(User user) throws SW360Exception {
-        // Check required fields
-        assertNotEmpty(user.getEmail());
-        // Set type
+        assertNotEmpty(user.getEmail(), "User email cannot be empty");
         user.setType(TYPE_USER);
-        // guarantee that `CommentMadeDuringModerationRequest` is never stored in the database as this is intended to be a transient field
         user.unsetCommentMadeDuringModerationRequest();
     }
 
+    /**
+     * Prepares a Vendor object for database insertion.
+     *
+     * @param vendor The Vendor to prepare
+     * @throws SW360Exception if validation fails
+     */
     public static void prepareVendor(Vendor vendor) throws SW360Exception {
-        // Check required fields
-        assertNotEmpty(vendor.getShortname(), "vendor short name cannot be empty!");
-        assertNotEmpty(vendor.getFullname(), "vendor full name cannot be empty!");
+        assertNotEmpty(vendor.getShortname(), "Vendor short name cannot be empty");
+        assertNotEmpty(vendor.getFullname(), "Vendor full name cannot be empty");
         assertValidUrl(vendor.getUrl());
-
-        // Check type
         vendor.setType(TYPE_VENDOR);
     }
 
+    /**
+     * Prepares a Component object for database insertion.
+     *
+     * @param component The Component to prepare
+     * @throws SW360Exception if validation fails
+     */
     public static void prepareComponent(Component component) throws SW360Exception {
-        // Check required fields
-        assertNotEmpty(component.getName());
-
-        // Check type
+        assertNotEmpty(component.getName(), "Component name cannot be empty");
         component.setType(TYPE_COMPONENT);
-
-        // Unset optionals
         component.unsetPermissions();
-
-        // Unset fields that do not make sense
         component.unsetReleases();
         component.unsetDefaultVendor();
-        // we might want to have a logic someday that the defaultVendorId will be set
-        // from defaultVendor if the latter one is set and the former one not - but
-        // until now the thought is that clients need to make sure that the
-        // defaultVendorId is set and that the component contains the vendor itself is
-        // just convenience
```

Comment:
Please do not just delete random things. These are important documentation comments.